### PR TITLE
Makefile: replace INCDIR to INCDIR_PREFIX

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
For more information see:
https://github.com/apache/incubator-nuttx/pull/7076

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

File: system/uorb/Make.defs [hunk 1]
```diff
@@ -20,8 +20,8 @@
...
 ifneq ($(CONFIG_UORB),)
 CONFIGURED_APPS += $(APPDIR)/system/uorb
-CFLAGS          += ${shell $(INCDIR) "$(CC)" \
-                   $(APPDIR)/system/uorb/}
-CXXFLAGS        += ${shell $(INCDIR) "$(CC)" \
-                   $(APPDIR)/system/uorb/}
+CFLAGS          += ${INCDIR_PREFIX}\
+                   $(APPDIR)/system/uorb/
+CXXFLAGS        += ${INCDIR_PREFIX}\
+                   $(APPDIR)/system/uorb/
```
Comment (xiaoxiang781216) [inline system/uorb/Make.defs:26]: merge into one line

File: interpreters/wamr/Make.defs [hunk 2]
```diff
@@ -21,9 +21,9 @@
 ifneq ($(CONFIG_INTERPRETERS_WAMR),)
 CONFIGURED_APPS += $(APPDIR)/interpreters/wamr
 
-CFLAGS += ${shell $(INCDIR) "$(CC)" \
-    $(APPDIR)/interpreters/wamr/wamr/core/iwasm/include}
-CXXFLAGS += ${shell $(INCDIR) "$(CC)" \
-    $(APPDIR)/interpreters/wamr/wamr/core/iwasm/include}
+CFLAGS += ${INCDIR_PREFIX}\
+    $(APPDIR)/interpreters/wamr/wamr/core/iwasm/include
+CXXFLAGS += ${INCDIR_PREFIX}\
```
Comment (xiaoxiang781216) [inline interpreters/wamr/Make.defs:26]: merge into one line

File: interpreters/quickjs/Make.defs [hunk 3]
```diff
@@ -22,9 +22,9 @@ ifneq ($(CONFIG_INTERPRETERS_QUICKJS),)
 CONFIGURED_APPS += $(APPDIR)/interpreters/quickjs
 
 # It allows `<quickjs/quickjs.h>` import.
-CFLAGS += ${shell $(INCDIR) "$(CC)" \
-    $(APPDIR)/interpreters/quickjs}
-CXXFLAGS += ${shell $(INCDIR) "$(CC)" \
-    $(APPDIR)/interpreters/quickjs}
+CFLAGS += ${INCDIR_PREFIX}\
```
Comment (xiaoxiang781216) [inline interpreters/quickjs/Make.defs:25]: ditto

File: interpreters/lua/Make.defs [hunk 4]
```diff
@@ -23,9 +23,9 @@ CONFIGURED_APPS += $(APPDIR)/interpreters/lua
 
 # Enable <lua.h> and <lauxlib.h> includes.
 
-CFLAGS += ${shell $(INCDIR) "$(CC)" \
-	$(APPDIR)/interpreters/lua/lua-$(CONFIG_INTERPRETER_LUA_VERSION)/src}
-CXXFLAGS += ${shell $(INCDIR) "$(CC)" \
-	$(APPDIR)/interpreters/lua/lua-$(CONFIG_INTERPRETER_LUA_VERSION)/src}
+CFLAGS += ${INCDIR_PREFIX}\
```
Comment (xiaoxiang781216) [inline interpreters/lua/Make.defs:26]: ditto

File: gpsutils/minmea/Make.defs [hunk 5]
```diff
@@ -20,6 +20,6 @@
 
 ifneq ($(CONFIG_GPSUTILS_MINMEA_LIB),)
 CONFIGURED_APPS += $(APPDIR)/gpsutils/minmea
-CFLAGS += ${shell $(INCDIR) "$(CC)" \
-    $(APPDIR)/gpsutils/minmea}
+CFLAGS += ${INCDIR_PREFIX}\
+    $(APPDIR)/gpsutils/minmea
```
Comment (xiaoxiang781216) [inline gpsutils/minmea/Make.defs:24]: merge into one line

File: interpreters/duktape/Make.defs [hunk 6]
```diff
@@ -22,9 +22,9 @@ ifneq ($(CONFIG_INTERPRETERS_DUKTAPE),)
 CONFIGURED_APPS += $(APPDIR)/interpreters/duktape
 
 # It allows `<duktape/duktape.h>` import.
-CFLAGS += ${shell $(INCDIR) "$(CC)" \
-    $(APPDIR)/interpreters/duktape/duktape/src-noline}
-CXXFLAGS += ${shell $(INCDIR) "$(CC)" \
-    $(APPDIR)/interpreters/duktape/duktape/src-noline}
+CFLAGS += ${INCDIR_PREFIX}\
```
Comment (xiaoxiang781216) [inline interpreters/duktape/Make.defs:25]: merge

**Review by jerpelea — APPROVED** [review]

File: interpreters/duktape/Make.defs (hunk 6 above)
Comment (Gary-Hobson) [reply interpreters/duktape/Make.defs]: all done ~

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**